manila/network/neutron/api.py

```diff
@@ -172,8 +173,14 @@ def delete_port(self, port_id):
         try:
             self.client.delete_port(port_id)
         except neutron_client_exc.NeutronClientException as e:
-            raise exception.NetworkException(code=e.status_code,
-                                             message=e.message)
+            # Not found means already deleted, we are usually fine
+            if e.status_code == 404:
+                msg = _('Neutron error deleting port %(port)s: %(error)s')
+                msg_args = {'port': port_id, 'error': e.message}
+                LOG.debug(msg, msg_args)
+            else:
+                raise exception.NetworkException(code=e.status_code,
+                                                 message=e.message)
```

Comment:
looks ok, but would be better to put the logic to where this function is called
https://github.com/sapcc/manila/blob/089594d0793c304ef44be719601cc5c61802b7a6/manila/network/neutron/neutron_network_plugin.py#L341-L344
exceptions are handled at the same place

Reply:
bc of the following else block, current implementaion could delete the network allocations before the ports